roles/confluent.zookeeper/tasks/dynamic_groups.yml

```diff
@@ -16,6 +16,17 @@
       list | first | regex_search('[0-9]+(.[0-9]+)+') }}"
   when: installation_method == "archive"
 
+- name: Load zookeeper.properties
+  slurp:
+    src: "{{ zookeeper.config_file }}"
+  register: slurped_properties
+
+- name: Set Current zookeeper_ssl_enabled, Create Backup
+  set_fact:
+    zookeeper_ssl_enabled_backup: "{{ zookeeper_ssl_enabled }}"
```

Comment:
I am not sure if I like this "backup" variable. 😄 Have you tried setting the "slurped" value of `zookeeper_ssl_enabled` directly on the "Get Leader/Follower" task? I am not sure if Ansible supports that kind of variable scoping? I am programmer, so I would expect it to work, but Ansible is not a programming language.....

Reply:
oh thats a great idea! is this what you are suggesting?
```
- name: Get Leader/Follower
  shell: >
    set -o pipefail &&
      {{ zookeeper_health_check_command }} | grep Mode
  args:
    executable: /bin/bash
  vars:
    archive_version: "{{ zookeeper_current_version | default(confluent_package_version) }}"
    zookeeper_ssl_enabled: "{{ (slurped_properties.content|b64decode).split('\n') |
      select('match', '^ssl.clientAuth=need') | length > 0 }}"
  register: leader_query
  changed_when: false
  check_mode: false
```

I can test it out

Reply:
Yes, exactly what I suggested. But needs to be tested. I suspect Ansible does not work as I expect it....

Reply:
It worked, thanks for the good recommendation